auth-web/src/components/auth/PasscodeForm.vue

```diff
@@ -81,79 +81,79 @@
-                { 'access_token': response.data.access_token,
-                  'refresh_token': response.data.refresh_token,
-                  'registries_trace_id': response.data['registries-trace-id'] }
-              )
-              IframeServices.emit(this.$refs.iframeContent.contentWindow, msg)
-              sessionStorage.KEYCLOAK_TOKEN = response.data.access_token
-              sessionStorage.KEYCLOAK_REFRESH_TOKEN = response.data.refresh_token
-              sessionStorage.REGISTRIES_TRACE_ID = response.data['registries-trace-id']
-              setTimeout(() => {
-                window.location.href = this.VUE_APP_COPS_REDIRECT_URL
-              }, 500)
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+import { getModule } from 'vuex-module-decorators'
+import BusinessModule from '../../store/modules/business'
+import configHelper from '../../util/config-helper'
+import iframeServices from '../../services/iframe.services'
+
+@Component
+export default class PasscodeForm extends Vue {
+  showPasscode = false
+  showSpinner = false
+  noPasscodeDialog = false
+  loginError = ''
+  valid = false
+  VUE_APP_COPS_REDIRECT_URL = configHelper.getValue('VUE_APP_COPS_REDIRECT_URL')
+  entityNumRules = [
+    v => !!v || 'Incorporation Number is required'
+  ]
+  entityPasscodeRules = [
+    v => !!v || 'Passcode is required',
+    v => v.length >= 9 || 'Passcode must be exactly 9 digits'
+  ]
+  businessStore = getModule(BusinessModule)
+
+  businessNumber: string = ''
+  passcode: string = ''
+
+  private isFormValid (): boolean {
+    return (this.$refs.form as Vue & { validate: () => boolean }).validate()
+  }
+
+  private getIFrameContent (): Window {
+    return (this.$refs.iframeContent as Vue & { contentWindow: Window }).contentWindow
+  }
+
+  login () {
+    if (this.isFormValid()) {
+      this.showSpinner = true
+      this.businessStore.login({ businessNumber: this.businessNumber, passCode: this.passcode })
+        .then(response => {
+          // set token and store in storage
+          // TODO: Once iframe token emit is no longer needed,
+          // this should be moved to the login Vuex action
+          const msg = JSON.stringify(
+            { 'access_token': response.data.access_token,
+              'refresh_token': response.data.refresh_token,
+              'registries_trace_id': response.data['registries-trace-id']
             }
-          })
-          .catch(response => {
-            this.loginError = this.$t('loginFailedMessage')
-          })
-      }
+          )
+          iframeServices.emit(this.getIFrameContent(), msg)
+          sessionStorage.KEYCLOAK_TOKEN = response.data.access_token
+          sessionStorage.KEYCLOAK_REFRESH_TOKEN = response.data.refresh_token
+          sessionStorage.REGISTRIES_TRACE_ID = response.data['registries-trace-id']
+
+          // attempt to load business
+          this.businessStore.loadBusiness(this.businessNumber)
```

Comment:
Yeah, for MVP, we'll create a business record on log in (via passcode) if we don't already have a record.  We'll always redirect to contact UI if there is no contact info present.

Post MVP will likely be a different since we won't be logging in with passcode at that point